### PR TITLE
Fetch::Config: existing attributes and parameters should precede resp parameters and configvalues

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

**Comment by stdweird** [comment]
found this out while looking to quick-fix https://github.com/quattor/aii/issues/193, by passing `TABCOMPLETION => 0` in the `new()`
not sure if there's any actual issue that this solves

**Comment by ned21** [comment]
I don't follow what the problem is here so can't comment on whether this is a good fix or not. :(

**Comment by stdweird** [comment]
passing `0` to attribute with e.g. `->new(ABC => 0)` does not precede config value due to code like `$self->{uc($p)} ||= $param->{uc($p)}`.

File: src/main/perl/Fetch/Config.pm [hunk 1]
```diff
@@ -40,9 +40,12 @@ sub _config
 
     my @keys = qw(tmp_dir context_url);
     push(@keys, @CFG_KEYS);
-    foreach my $p (@keys) {
+    foreach my $k (@keys) {
+        my $K = uc($k);
```
Comment (ned21) [inline src/main/perl/Fetch/Config.pm]: Opinions on $k and $K -- too subtle and easy to miss/typo? Can't say I like $uck either. $ucK might work? Or, tbh I might be tempted to spell out uc($k) every time - it's not saving very much typing/reading to use a separate variable

**Comment by stdweird** [comment]
@ned21 remark fixed

**Pull request merged**